modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -330,12 +330,23 @@ object CurrencySnapshotAcceptanceManager {
       lastGlobalSnapshotEpochProgress = lastSyncGlobalSnapshot.epochProgress
       lastGlobalSnapshotOrdinal = lastSyncGlobalSnapshot.ordinal
 
+      globalSyncView = maybeLastGlobalSyncView match {
+        case Some(lastGlobalSyncView) =>
+          if (lastGlobalSyncView.ordinal < lastSyncGlobalSnapshot.ordinal) {
+            GlobalSyncView(lastSyncGlobalSnapshot.ordinal, lastSyncGlobalSnapshot.hash, lastSyncGlobalSnapshot.epochProgress)
+          } else {
+            lastGlobalSyncView
+          }
+        case None => GlobalSyncView(lastSyncGlobalSnapshot.ordinal, lastSyncGlobalSnapshot.hash, lastSyncGlobalSnapshot.epochProgress)
+      }
```

Comment:
minor suggestion but continuous pattern matching (the boxing / unboxing) may eventually impact performance so it would be my preference for us to use functional approach instead

```scala
globalSyncView = maybeLastGlobalSyncView
  .filter(_.ordinal >= lastSyncGlobalSnapshot.ordinal)
  .getOrElse(
    GlobalSyncView(
      lastSyncGlobalSnapshot.ordinal, 
      lastSyncGlobalSnapshot.hash, 
      lastSyncGlobalSnapshot.epochProgress
    )
  )
```